Add -F option to find method

krafczyk opened the pull request:
Occationally, a package can be built with different compilers for
some components. In these situations, it helps to be able to see
the full compiler version information. -F has been added to find
to accomplish this task.

adamjstewart commented:
Can you add this new flag to Spack's Bash tab-completion?
https://github.com/LLNL/spack/blob/develop/share/spack/spack-completion.bash#L360

krafczyk commented:
@adamjstewart done.

krafczyk commented:
@becker33, @tgamblin Is there a possibility of getting this merged?

scheibelp commented on lib/spack/spack/cmd/__init__.py:
```diff
@@ -219,7 +220,13 @@ def get_arg(name, default=None):
     nfmt = '.' if namespace else '_'
-    ffmt = '$%+' if flags else ''
+    ffmt = ''
+    if full_compiler or flags:
```
I think the ```--show-flags``` option should be removed and when you specify ```-F``` you get the version and the flags. At the moment IMO it's confusing when ```-F``` doesn't print the flags too and you also need ```-f```. If the option to include the compiler version was ```--show-compiler-version``` it would be fine.

krafczyk commented:
@scheibelp Okay, how about this, I removed `-F`, and changed it to `--show-compiler-version`. Now if you specify `--show-compiler-version` it will show flags and version, and if just `-f`, it'll just show flags.

scheibelp commented on share/spack/spack-completion.bash:
```diff
@@ -363,9 +363,9 @@ function _spack_find {
-                    -L --very-long -f --show-flags -e --explicit
-                    -E --implicit -u --unknown -m --missing -v --variants
-                    -M --only-missing -N --namespace" -- "$cur"
+                    -L --very-long -f --show-flags -F --show-full-compiler
```
If you think the option name should remain ```--show-compiler-version``` then this should change. That being said IMO if ```--show-compiler-version``` is going to show the flags and version I'd rather it was called ```--show-full-compiler``` anyway, so I'd prefer just changing the option name back in ```find```.

krafczyk replied:
I've gone with your suggestion, and changed the option to `--show-full-compiler`.

krafczyk commented:
@scheibelp I've gone with your changes, can you have another look at this PR?

scheibelp commented:
Thanks!

krafczyk commented:
@scheibelp Thanks!
